File: 02 - api/javaScrpit/climate/src/service/climate_service.js
import axios from "axios";
import dotenv from "dotenv";

dotenv.config({ path: "./src/.env" });

class ClimateService {
  constructor() {
    this.api_url = "https://apiadvisor.climatempo.com.br/api/v1";
    this.api_key = process.env.API_KEY;
  }

  async consultarUrl(endpoint) {
    const url = `${this.api_url}/${endpoint}?token=${this.api_key}`;
    try {
      const response = await axios.get(url);
      return response.data;
    } catch (error) {
      return {
        error: `Falha ao obter dados. Código de status: ${error.response.status}`,
      };
    }
  }

  async getClimate(country) {
    return this.consultarUrl(`anl/synoptic/locale/${country}`);
  }

  async getClimateRain(id) {
    return this.consultarUrl(`climate/temperature/locale/${id}`);
  }
}

export default ClimateService;

File: 02 - api/python/api/climate/service/climate_service.py
# climate/services/climate_service.py
import os
import requests
from dotenv import load_dotenv

load_dotenv()

class ClimateService:
    def __init__(self):
        self.api_url =  "https://apiadvisor.climatempo.com.br/api/v1";
        self.api_key = os.getenv('API_KEY')

    def consultar_url(self, endpoint):
        url = f"{self.api_url}/{endpoint}?token={self.api_key}"
        response = requests.get(url)
        if response.status_code == 200:
            return response.json()
        else:
            return {"error": f"Falha ao obter dados. Código de status: {response.status_code}"}

    def get_climate(self, country):
        return self.consultar_url(f"anl/synoptic/locale/{country}")

    def get_climate_rain(self, id):
        return self.consultar_url(f"climate/temperature/locale/{id}")